application/libraries/Api/Command/V1/SurveyPatch/OpHandlerLanguageSettingsUpdate.php

```diff
@@ -96,21 +96,23 @@ public function handle(OpInterface $op): void
         $languageSettings = $diContainer->get(
             LanguageSettings::class
         );
-        $data = $this->transformer->transformAll(
-            $op->getProps(),
-            [
-                'operation' => $op->getType()->getId(),
-                'entityId'  => $op->getEntityId(),
-                'sid'       => $this->getSurveyIdFromContext($op)
-            ]
-        );
-        if (empty($data)) {
-            $this->throwNoValuesException($op);
+        foreach ($op->getProps() as $languageCode => $props) {
+            $data = $this->transformer->transformAll(
```

Comment:
transformAll() is meant to always take collections of entities as a whole and deal with it. Opposed to the function transform() which is mean to handle single entities
Looping over the props, splitting the collection up, would be the job of the transformAll function itself.

Reply:
I guess the whole issue is just an issue because the frontend always uses approach 2 (as seen in the docblock of \LimeSurvey\Api\Command\V1\SurveyPatch\OpHandlerLanguageSettingsUpdate::handle)?

Reply:
The id can be anything, it's not broken but also not very clean to pass language there.
The survey id is always passed in the "context" anyway.

If the props come in as 
```
{
            en: {
                title: "kdsjdskl"
           }
        }
```
they are considered a collection, even if the collection only consists of one element.
so transformAll should handle it.